test/tests/functional/pbs_standing_resv_end_hook.py

```diff
@@ -0,0 +1,172 @@
+"""
+        self.hook_name = 'resvend_hook'
+        attrs = {'event': 'resv_end'}
+        self.server.create_import_hook(
+            self.hook_name, attrs, hook_script, overwrite=True)
+
+        self.server.manager(
+            MGR_CMD_SET, SERVER, {
+                'log_events': 2047}, expect=True)
+
+    def submit_and_confirm_resv(
+            self,
+            offset,
+            duration,
+            select="1:ncpus=1",
+            rrule='FREQ=MINUTELY;COUNT=2'):
+        """ Method to submit and confirm the standing reservation """
+
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using Asia/Kolkata')
+            tzone = 'Asia/Kolkata'
+
+        start_time = time.time() + offset
+        end_time = start_time + duration
+
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select,
+            ATTR_resv_rrule: rrule,
+            ATTR_resv_timezone: tzone}
+
+        rid = self.server.submit(Reservation(TEST_USER, attrs))
+
+        resv_attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, resv_attrs, id=rid)
+
+        return rid
+
+    def test_delete_resv(self):
+        """ Test to delete one of the occurrences of a confirmed
+            standing reservation and verify the resvend hook. """
+
+        offset = 3
+        duration = 10
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        time.sleep(10)
+
+        msg = "Hook;Server@" + self.server.hostname.lower() + \
+            ";Reservation occurrence - %s" % '1'
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+        self.logger.info(
+            "Reservation end hook ran for first occurrence of a standing reservation")
+
+        time.sleep(5)
+        self.server.delete(rid)
+        msg = "Hook;Server@" + self.server.hostname.lower() + \
+            ";Reservation occurrence - %s" % '2'
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+
+    def test_end_resv(self):
+        """ Test to end all the occurrences of the confirmed
+            standing reservation and verify the resvend hook
+            runs for all the occurrences. """
+
+        offset = 3
+        duration = 10
+        rid = self.submit_and_confirm_resv(offset, duration)
+
+        time.sleep(10)
+
+        msg = "Hook;Server@" + self.server.hostname.lower() + \
+            ";Reservation occurrence - %s" % '1'
+        self.server.log_match(msg, tail=True, interval=2, max_attempts=30)
+        self.logger.info(
+            "Reservation end hook ran for first occurrence of a standing reservation")
+
+        # Sleep for one minute as the reservation occurrence is for minutely
+        time.sleep(50)
```

Comment:
Add a log message saying that you're sleeping for 50s and why you are (same with other tests)